### PR TITLE
Fixing the assignment toggle for when a different container is open. …

### LINKED ISSUE #611: "Always open in this container" option is checked in other containers even if the website is not assigned to them
[Affected versions]:
- Firefox 54.0.1 and up
- Containers 2.3.0 

[Affected Platforms]:
- All Windows
- All Mac
- All Linux

[Prerequisites]:
- Latest Containers add-on was previously installed on a clean profile.

[Steps to reproduce]:
1. Open the browser with the profile from prerequisites and navigate to a website in a container tab. (eg. www.reddit.com)
2. Right click on a link and choose the "Always open in this container" option.
3. Open a new container tab, different from the first, and navigate to the same website.
4. Choose to continue in the current container tab.
5. Right click on a link and observe the "Always open in this container" option.

[Expected result]:
- Option is unchecked.

[Actual result]:
- Option is checked.

[Notes]:
- Unchecking the option in a different container also shows an incorrect "Successfully removed site to open in **this** container" message.
- Attached a screen recording of the issue:
always open option is in the same state for different containers

### PR DESCRIPTION
**Pull request by jonathanKingston**
…Fixes #611

### CODE REVIEW

File: webextension/background.js [hunk 1]
```diff
@@ -254,7 +254,9 @@ const assignManager = {
     // ✓ This is to mitigate https://bugzilla.mozilla.org/show_bug.cgi?id=1351418
     let prefix = "   "; // Alignment of non breaking space, unknown why this requires so many spaces to align with the tick
     let menuId = this.MENU_ASSIGN_ID;
-    if (siteSettings) {
+    const userContextId = this.getUserContextIdFromCookieStore(tab);
```
Comment (groovecoder) [inline webextension/background.js]: nit: call it `tabUserContextId` so it's more clear.

File: webextension/js/popup.js [hunk 2]
```diff
@@ -482,16 +485,16 @@ Logic.registerPanel(P_CONTAINERS_LIST, {
...
     const currentTabElement = document.getElementById("current-tab");
     const assignmentCheckboxElement = document.getElementById("container-page-assigned");
+    const userContextId = Logic.userContextId(currentTab.cookieStoreId);
     assignmentCheckboxElement.addEventListener("change", () => {
-      const userContextId = Logic.userContextId(currentTab.cookieStoreId);
       Logic.setOrRemoveAssignment(currentTab.id, currentTab.url, userContextId, !assignmentCheckboxElement.checked);
     });
     currentTabElement.hidden = !currentTab;
-    this.setupAssignmentCheckbox(false);
+    this.setupAssignmentCheckbox(false, userContextId);
```
Comment (groovecoder) [inline webextension/js/popup.js]: nit: call it `currentTabUserContextId` to be more clear.

File: webextension/js/popup.js [hunk 3]
```diff
@@ -482,16 +485,16 @@ Logic.registerPanel(P_CONTAINERS_LIST, {
...
     if (currentTab) {
       const identity = await Logic.identity(currentTab.cookieStoreId);
       const siteSettings = await Logic.getAssignment(currentTab);
-      this.setupAssignmentCheckbox(siteSettings);
+      this.setupAssignmentCheckbox(siteSettings, userContextId);
```
Comment (groovecoder) [inline webextension/js/popup.js]: nit: call it `currentTabUserContextId` to be more clear.

File: webextension/js/popup.js [hunk 4]
```diff
@@ -482,16 +485,16 @@ Logic.registerPanel(P_CONTAINERS_LIST, {
     const currentTab = await Logic.currentTab();
     const currentTabElement = document.getElementById("current-tab");
     const assignmentCheckboxElement = document.getElementById("container-page-assigned");
+    const userContextId = Logic.userContextId(currentTab.cookieStoreId);
```
Comment (groovecoder) [inline webextension/js/popup.js]: nit: call it `currentTabUserContextId` to be more clear.

File: webextension/background.js (hunk 1 above)
Comment (jonathanKingston) [inline webextension/background.js]: Done

Comment (groovecoder) [inline webextension/background.js]: ? I don't see it updated here on GitHub ...

File: webextension/js/popup.js (hunk 2 above)
Comment (jonathanKingston) [inline webextension/js/popup.js]: Done

File: webextension/js/popup.js (hunk 3 above)
Comment (jonathanKingston) [inline webextension/js/popup.js]: Done

File: webextension/js/popup.js (hunk 4 above)
Comment (jonathanKingston) [inline webextension/js/popup.js]: Done

**Comment by jonathanKingston** [comment]
Ah just saw your comment:

> nits can be fixed before or after merge.

Fixed hopefully :)

**Pull request merged**